graphs/graph_modules/data_fetcher.py

```
@@ -0,0 +1,371 @@
+            
+            api_params = {
+                "apikey": self.config["TAUTULLI_API_KEY"],
+                "cmd": cmd,
+                **params
+            }
+
+            # Ensure all params are strings for the API call
+            api_params = {k: str(v) for k, v in api_params.items() if v is not None}
+
+            async with aiohttp.ClientSession() as session:
+                async with session.get(self.config["TAUTULLI_URL"], params=api_params) as response:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)